lib/ui/src/app.tsx

```diff
@@ -47,15 +47,20 @@ const App = React.memo<AppProps>(
           {
             key: 'settings',
             render: () => <SettingsPages />,
-            route: (({ children }) => (
-              <Route path="/settings" startsWith>
-                {children}
-              </Route>
-            )) as FunctionComponent,
+            route: (({ children }) => {
+              if (viewMode === 'settings') {
+                return (
+                  <Route path="/settings" startsWith>
+                    {children}
+                  </Route>
```

Comment:
Isn't this redundant?
https://github.com/storybookjs/storybook/blob/d0b533db1178abf32df0796d6da17b73f82a35e7/lib/router/src/router.tsx#L104-L113

The Route component won't render it's children if the route doesn't match..

Reply:
@ndelangen I don't think so? If you pull down and test the branch locally, are you able to set a story's id to `settings`, view the story, and the controls panel?

Reply:
Hmmm, it's supposed to only match on `/settings` not on `/story/settings-...`.

The true bug here is that `<Route path="/settings" startsWith>` doesn't actually behave correctly.